tests/director/gateway-integration/automatic_status_update_test.go

```diff
@@ -0,0 +1,182 @@
+package gateway_integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/str"
+
+	"github.com/kyma-incubator/compass/tests/director/pkg/ptr"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/director/pkg/gql"
+	"github.com/kyma-incubator/compass/tests/director/pkg/idtokenprovider"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAutomaticStatusUpdate(t *testing.T) {
+	domain := os.Getenv("DOMAIN")
+	require.NotEmpty(t, domain)
+	tenant := os.Getenv("DEFAULT_TENANT")
+	require.NotEmpty(t, tenant)
+	ctx := context.Background()
+
+	t.Log("Get Dex id_token")
+	dexToken, err := idtokenprovider.GetDexToken()
+	require.NoError(t, err)
+
+	dexGraphQLClient := gql.NewAuthorizedGraphQLClient(dexToken)
+
+	t.Run("Test status update as static user", func(t *testing.T) {
+
+		t.Log("Register application as Static User")
+		appInput := graphql.ApplicationRegisterInput{
+			Name:         "app-static-user",
+			ProviderName: ptr.String("compass"),
+		}
+		app, err := registerApplicationWithinTenant(t, ctx, dexGraphQLClient, tenant, appInput)
+		require.NoError(t, err)
+
+		defer unregisterApplication(t, ctx, dexGraphQLClient, tenant, app.ID)
+
+		assert.Equal(t, graphql.ApplicationStatusConditionInitial, app.Status.Condition)
+
+		status := graphql.ApplicationStatusConditionFailed
+		t.Log("Update the application")
+		appUpdateInput := graphql.ApplicationUpdateInput{
+			Description:     str.Ptr("New description"),
+			StatusCondition: &status,
+		}
+		appUpdated, err := updateApplicationWithinTenant(t, ctx, dexGraphQLClient, tenant, app.ID, appUpdateInput)
+		require.NoError(t, err)
+
+		t.Log("Ensure the status condition")
+		assert.Equal(t, graphql.ApplicationStatusConditionFailed, appUpdated.Status.Condition)
+	})
+
+	t.Run("Test status update as Integration System", func(t *testing.T) {
```

Comment:
I wonder if tests for IS and users are needed?
Because we test the flow in unit tests.